### PR TITLE
Windows integration tests

### PR DESCRIPTION
**Pull request by jsteinich**
Aiming to fix https://github.com/hashicorp/terraform-cdk/issues/201

Planning to convert most test scripts to JS and then run on a GitHub Windows runner.

### CODE REVIEW

**Comment by skorfmann** [comment]
I think I might have misunderstood your comment [here](https://github.com/hashicorp/terraform-cdk/issues/201#issuecomment-663525555), sorry!

> Do you have a preference between platform specific scripts vs writing something in JS/TS/Python that would be cross platform?

For the integration test, a script similar [to this](https://github.com/hashicorp/terraform-cdk/blob/8a5d26bf8b84cf19967f4bdb773fc405eca30a05/test/test-typescript-deploy/test.sh) would be good.

**Comment by jsteinich** [comment]
Turns out this is currently blocked on https://github.com/hashicorp/terraform/issues/25068 as the apply fails with no useful information other than error code of 1.

**Comment by jsteinich** [comment]
> Turns out this is currently blocked on [hashicorp/terraform#25068](https://github.com/hashicorp/terraform/issues/25068) as the apply fails with no useful information other than error code of 1.

I put in a workaround for this since the fix is only in the 0.13 beta version.

@skorfmann Would you like me to convert the other tests to also run on Windows? The Windows tests run in parallel to the linux ones, so it should take about the same wall clock time to run. I do think the python test is likely to fail at this point, so I'd probably skip that one.

If just this one test is good, then there's just a small amount of cleanup I'd like to do before merging.

One other note is that I'm only running the Windows tests against the 0.12.X version currently. It shouldn't be too difficult to add in 0.13 if desired.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -28,8 +28,50 @@ jobs:
...
+          path: dist
       - name: integration tests
         run: yarn integration
         env:
           TERRAFORM_CLOUD_TOKEN: ${{ secrets.TERRAFORM_CLOUD_TOKEN }}
           TERRAFORM_BINARY_NAME: "terraform${{ matrix.terraform }}"
+
+  windows_integration:
+    runs-on: windows-latest
+    needs: build
```
Comment (skorfmann) [inline .github/workflows/build.yml:61]: Can we make this a matrix build as well?

**Comment by skorfmann** [comment]
Pretty cool - thanks again 👍

**Pull request merged**